src/handler/promote.ts

```diff
@@ -0,0 +1,99 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import 'dotenv/config';
+import { cloneDeep } from 'lodash';
+import { TechRecordGet } from '../models/post';
+import { PromoteRecordRequestBody } from '../models/promote';
+import { SearchCriteria } from '../models/search';
+import { archiveOldCreateCurrentRecord, getBySystemNumberAndCreatedTimestamp, searchByCriteria } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { validatePromoteErrors } from '../validators/promote';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Promote end point called');
+
+  const promoteErrors = validatePromoteErrors(event);
+
+  if (promoteErrors) {
+    return addHttpHeaders(promoteErrors);
+  }
+
+  const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber ?? '');
+  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
+  const userDetails = getUserDetails(event.headers.Authorization ?? '');
+  const { reasonForPromoting } = JSON.parse(event.body as string) as PromoteRecordRequestBody;
+
+  logger.info(`Get from database with systemNumber ${systemNumber} and timestamp ${createdTimestamp}`);
+
+  const provisionalRecord: TechRecordGet = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp) as TechRecordGet;
+
+  logger.debug(`result is: ${JSON.stringify(provisionalRecord)}`);
+
+  if (!provisionalRecord || !Object.keys(provisionalRecord).length) {
+    return addHttpHeaders({
+      statusCode: 404,
+      body: `No record found matching systemNumber ${systemNumber} and timestamp ${createdTimestamp}`,
+    });
+  }
+
+  if (provisionalRecord.techRecord_statusCode !== 'provisional') {
+    return addHttpHeaders({
+      statusCode: 400,
+      body: 'Record provided is not a provisional record so cannot be promoted.',
+    });
+  }
+
+  const allVehicleRecords = await searchByCriteria(SearchCriteria.SYSTEM_NUMBER, provisionalRecord.systemNumber);
+  const currentResult = allVehicleRecords.filter((searchResult) => searchResult.techRecord_statusCode === 'current');
+
+  logger.debug(`Current result ${JSON.stringify(currentResult)}`);
+
+  const recordsToArchive: TechRecordGet[] = [];
+
+  if (currentResult.length > 0) {
+    const currentRecord = await getBySystemNumberAndCreatedTimestamp(currentResult[0].systemNumber, currentResult[0].createdTimestamp) as TechRecordGet;
+    currentRecord.techRecord_statusCode = 'archived';
+    currentRecord.techRecord_lastUpdatedAt = new Date().toISOString();
+    currentRecord.techRecord_lastUpdatedByName = userDetails.username;
+    currentRecord.techRecord_lastUpdatedById = userDetails.msOid;
```

Comment:
I think @shivangidas's PR has some helped functions for the lastUpdatedAt etc... logic. I don't really know how to go implementing them in this PR as well but would be good to standardise the audit logic

Reply:
Yeah I've seen that... I think we put hers in first then I just redo a load of stuff tbh